### DISCUSSION TITLE
Question for create md files from extra vars

### DISCUSSION THREAD

**Discussion by alanlivio** [fralau/mkdocs-macros-plugin#132] [Ideas]
I know that `mkdcos-macros-plugin` enable the use of the `extra` variables from the mkdocs.yaml
```yaml
extra:
   data_for_page1
   data_for_page2
```
Such variables can be in existing markdown files, e.g., page.md uses `extra.data_for_page1` and page2.md uses `extra.data_for_page2`.

I am wondering if it is possible to use such variables to make mkdcos create the page1.md/page2.md files itself.

**Comment by fralau** [comment]
You mean, dynamically generating _new_ markdown pages? 🤔 

Interesting idea. Is the number of pages variable or static? Could you perhaps elaborate on your use case?

**Comment by alanlivio** [comment]
> You mean, dynamically generating new markdown pages? 🤔

Yes. But defined from some yaml.

>Interesting idea. Is the number of pages variable or static? Could you perhaps elaborate on your use case?

I think that may depend on the list of `extra` variables. 
Image the next mkdocs.yaml and I will create one page for each given icon/url.

```yml
extra:
   new_pages: 
      -  icon: "<url>"
         url: "<url>"
      -  icon: "<url>"
         url: "<url>"
```

**Comment by fralau** [comment]
Perhap it is possible to use mkdocs-macros for this, but I had not thought about it.

To avoid "reinventing the wheel", I would look at solutions that have already been implemented. For a dynamic behavior with a variable number pages, I would rather use an existing plugin: [awesome pages](https://github.com/lukasgeiter/mkdocs-awesome-pages-plugin). Would it do what your are expecting?

**Reply by alanlivio** [reply]
Hi @fralau. Thank you for the discussion.
I checked awesome-pages and it does more ordering/organzing pages from from .md files but not create them.
I still think that it will be a nice feature for the mkdocs-macros.

**Comment by fralau** [comment]
I am not saying that I hold the truth: but I would apply a [separation of concerns.](https://simplicable.com/new/separation-of-concerns), by generating your pages before launching the `mkdocs build` command. 

If we consider the document generation process, I would do it this way:

```mermaid
flowchart LR
   A[Generate\nmarkdown pages] --> F
   subgraph mkdocs
   F[awesomepages] --> B[macros] --> C[render pages] --> D[serve/build]
   end
```

You could generate pages containing macros commands, so that they could be converted later in the process. 

Integrating the two in a single script would be easy.

**Reply by alanlivio** [reply]
Hi @fralau thanks for the didactic explanation. I really appreciate that. I will evaluate it in my scenario.
